go/service/identify.go

```diff
@@ -147,8 +149,234 @@ func (h *IdentifyHandler) resolveUserOrTeam(ctx context.Context, arg string) (u
 	return res.UserOrTeam(), nil
 }
 
+func (h *IdentifyHandler) resolveUser(ctx context.Context, assertion string) (u keybase1.User, err error) {
+
+	res := h.G().Resolver.ResolveFullExpressionNeedUsername(ctx, assertion)
+	err = res.GetError()
+	if err != nil {
+		return u, err
+	}
+	u = res.User()
+	if !u.Uid.Exists() {
+		return u, fmt.Errorf("no resolution for: %v", assertion)
+	}
+	return u, nil
+}
+
 func (h *IdentifyHandler) ResolveIdentifyImplicitTeam(ctx context.Context, arg keybase1.ResolveIdentifyImplicitTeamArg) (res keybase1.ResolveIdentifyImplicitTeamRes, err error) {
-	return res, fmt.Errorf("TODO: implement ResolveIdentifyImplicitTeam")
+	ctx = libkb.WithLogTag(ctx, "RII")
+	defer h.G().CTrace(ctx, "IdentifyHandler#ResolveIdentifyImplicitTeam", func() error { return err })()
+
+	h.G().Log.CDebugf(ctx, "ResolveIdentifyImplicitTeam assertions:'%v'", arg.Assertions)
+
+	if len(arg.Assertions) == 0 {
+		return res, fmt.Errorf("empty assertions")
+	}
+
+	split := strings.Split(arg.Assertions, "#")
+	if len(split) > 2 {
+		return res, fmt.Errorf("too many reader divison ('#') in assertions: %v", arg.Assertions)
+	}
+
+	writerAssertions, err := h.parseAssertionList(ctx, split[0])
+	if err != nil {
+		return res, err
+	}
+
+	var readerAssertions []libkb.AssertionExpression
+	if len(split) >= 2 && len(split[1]) > 0 {
+		readerAssertions, err = h.parseAssertionList(ctx, split[1])
+		if err != nil {
+			return res, err
+		}
+	}
+
+	return h.resolveIdentifyImplicitTeamHelper(ctx, arg, writerAssertions, readerAssertions)
+}
+
+func (h *IdentifyHandler) resolveIdentifyImplicitTeamHelper(ctx context.Context, arg keybase1.ResolveIdentifyImplicitTeamArg,
```

Comment:
can you split this function up into smaller helpers?  would make it easier to read.

Reply:
pulled out identify part, maybe small enough now?